src/main/java/games/strategy/engine/data/AllianceTracker.java

```diff
@@ -94,4 +94,9 @@ protected void addToAlliance(final PlayerID player, final String allianceName) {
         .flatMap(Collection::stream)
         .collect(Collectors.toSet());
   }
+
+  @Override
+  public AllianceTracker clone() {
+    return new AllianceTracker(HashMultimap.create(alliances));
```

Comment:
This looks like a shallow clone.  Doesn't it need to be a deep clone since `PlayerID` is a `GameDataComponent`?

Note that this comment pretty much applies to all the new `clone()` methods in this PR.  If there exists, even transitively, a relationship between a `Cloneable` object and a `GameData`, a deep clone will most likely need to be performed, otherwise the clone will still retain a (transitive) reference to the original `GameData`.  And, in general, the intermediate objects will be shared between two or more `GameData`s, and that's going to lead to changes made to the cloned instance being visible in the original instance.